Tootsie Retro + 8b model card (#1236)
marin-community/marin: docs/model-cards/marin-8b.md

---
license: apache-2.0
datasets:
- allenai/dolmino-mix-1124
- allenai/olmo-mix-1124
- bigcode/starcoderdata
- EleutherAI/proof-pile-2
- hltcoe/megawika
- mlfoundations/dclm-baseline-1.0
- HuggingFaceTB/finemath
- marin-community/ar5iv-noproblem-markdown
- marin-community/ar5iv-warning-markdown
- marin-community/datashop-science-qa
- marin-community/stackexchange-markdown
- marin-community/wikipedia-markdown

> # REMINDER: when the instruct model should add dependencies on the instruct datasets and the base model.

Reviewer: I can't parse this sentence

Author: eh

  - [MegaWika](https://huggingface.co/datasets/hltcoe/megawika)
  - [peS2o](https://huggingface.co/datasets/allenai/peS2o)
  - (And most of the rest of it)
- [Dolmino-Mix-1124](https://huggingface.co/datasets/allenai/dolmino-mix-1124), including their versions of:
    - [FLAN](https://arxiv.org/abs/2109.01652)
    - [CodeSearchNet](https://arxiv.org/abs/1909.09436) (with OWM Filter)
    - [GSM8K](https://arxiv.org/pdf/2110.14168v1)
    - [MetaMath](https://arxiv.org/abs/2309.12284)
    - [MathCoder2 Synthetic](https://arxiv.org/abs/2310.03731)

And some new datasets:

- [Marin Markdownified StackExchange](https://huggingface.co/datasets/marin-community/stackexchange-markdown)
- [Marin Markdownified Wikipedia](https://huggingface.co/datasets/marin-community/wikipedia-markdown)
- [Marin Markdownified Ar5iv (No Problem)](https://huggingface.co/datasets/marin-community/ar5iv-noproblem-markdown)
- [Marin Markdownified Ar5iv (Warnings)](https://huggingface.co/datasets/marin-community/ar5iv-warning-markdown)
- [Marin Datashop Science QA](https://huggingface.co/datasets/marin-community/datashop-science-qa)

> The first three are licensed per their original licenses. The fourth is licensed under CC-BY-SA 4.0.

Reviewer: There are 5 items here...so to be clear, refer to the datasets by name

- [open-r1/OpenThoughts-114k-math](https://huggingface.co/datasets/open-r1/OpenThoughts-114k-math)
- [HuggingFaceTB/smoltalk](https://huggingface.co/datasets/HuggingFaceTB/smoltalk)
- [allenai/tulu-3-sft-mixture](https://huggingface.co/datasets/allenai/tulu-3-sft-mixture)
- [PrimeIntellect/verifiable-math-problems](https://huggingface.co/datasets/PrimeIntellect/verifiable-math-problems)

It is quite likely that we will release improved versions of this model in the future.

## Checkpoints

We release a large number of checkpoints.

### Base Model Checkpoints

Main Page: [marin-community/marin-8b-base](https://huggingface.co/marin-community/marin-8b-base)

| Name              | Training Tokens | Link                                                                                                       |
|-------------------|-----------------|------------------------------------------------------------------------------------------------------------|
| `main`            | 12.7T           | [marin-community/marin-8b-base](https://huggingface.co/marin-community/marin-8b-base/tree/main)            |

> | `kestrel`         | 2.7T            | [kestrel](https://huggingface.co/marin-community/marin-8b-base/tree/kestrel)                               |

Reviewer: Shouldn't these still have the `marin-community/` prefix?

> | `ocelot`          | 3.78T           | [kestrel](https://huggingface.co/marin-community/marin-8b-base/tree/ocelot)                                |

Reviewer 2: ```suggestion
| `ocelot`          | 3.78T           | [ocelot](https://huggingface.co/marin-community/marin-8b-base/tree/ocelot)                                |
```

| `starling`        | 12.4T           | [marin-community/marin-8b-base](https://huggingface.co/marin-community/marin-8b-base/tree/starling)        |
| `deeper-starling` | 12.7T           | [marin-community/marin-8b-base](https://huggingface.co/marin-community/marin-8b-base/tree/deeper-starling) |

`main` currently refers to `deeper-starling`. main currently refers to deeper-starling.
This may change in the future, but we will maintain compatibility at the architecture and tokenizer level,
so the model will remain drop-in compatible with existing tooling.
If you require a specific checkpoint, please use the `revision` argument.

### Instruct Model Checkpoints

Main Page: [marin-community/marin-8b-instruct](https://huggingface.co/marin-community/marin-8b-instruct)

| Name                    | SFT Tokens | Link                                                                                                                     |
|-------------------------|------------|--------------------------------------------------------------------------------------------------------------------------|

> | `main`                 | 5.3B       | [marin-community/marin-8b-instruct](https://huggingface.co/marin-community/marin-8b-instruct/tree/deeper-starling-05-15) |

Reviewer: nit: might as well align?

| `deeper-starling-05-15` | 5.3B       | [marin-community/marin-8b-instruct](https://huggingface.co/marin-community/marin-8b-instruct/tree/deeper-starling-05-15) |

> `main` currently refers to `deeper-starling-05-15`. This may change in the future, though we will maintain model compatibility. If you require a specific checkpoint, please use the `revision` argument.

Reviewer: may => will (probably)
what does model compatibility mean?  Models aren't really backward compatible

## Installation

Marin 8B uses the [Llama architecture](https://arxiv.org/abs/2302.13971) and as such should
work out-of-the-box with the [Hugging Face Transformers](https://huggingface.co/docs/transformers/index) library
and any other library that supports the Llama architecture.

We use a variant of the Llama 3 tokenizer: [stanford-crfm/marin-tokenizer](https://huggingface.co/stanford-crfm/marin-tokenizer/).

## Inference

> You can use Marin with the standard HuggingFace Transformers library:

Reviewer: ```suggestion
You can use Marin with the standard Hugging Face Transformers library:
```
Do this everywhere

```python
from transformers import AutoModelForCausalLM, AutoTokenizer
marin = AutoModelForCausalLM.from_pretrained("marin-community/marin-8b-base")
tokenizer = AutoTokenizer.from_pretrained("marin-community/marin-8b-base")
message = ["The Marin wind is"]
inputs = tokenizer(message, return_tensors='pt', return_token_type_ids=False)
response = marin.generate(**inputs, max_new_tokens=100, do_sample=True, top_k=50, top_p=0.95)
print(tokenizer.batch_decode(response, skip_special_tokens=True)[0])
```

We released a number of checkpoints of this model. To load a specific checkpoint, simply add the argument `revision`:

```bash
marin = AutoModelForCausalLM.from_pretrained("marin-community/marin-8b-base", revision="deeper-starling")
```

### Model Description

- **Developed by:** The Marin team at Stanford CRFM.
- **Model type:** a Transformer style autoregressive language model.

> - **Knowledge Cutoff:** ~July 2024

Reviewer: are we sure, given that we picked up new datasets since then?

- **Language(s) (NLP):** English
- **License:** The code and model are released under Apache 2.0.
- **Contact:** `dlwh at stanford.edu`

### Model Sources

- **Project Page:** https://marin.community
- **Repositories:**
    - Core repo (data and experiment management): https://github.com/marin-community/marin
    - Training code: https://github.com/stanford-crfm/levanter
- **Retrospective:** https://marin.readthedocs.io/en/latest/reports/marin-8b-retro.html
- **W&B Logs:** [Marin 8B](https://wandb.ai/stanford-mercury/marin/reports/Tootsie-8B---VmlldzoxMTY3MzU3OA)

## Evaluation

### Base Model Results

> We ran a suite of standard benchmarks to compare our model with [Llama 3.1 8B](https://huggingface.co/meta-llama/Meta-Llama-3.1-8B), and the open source 7-8B models [Olmo 2 7B](https://huggingface.co/allenai/OLMo-2-1124-7B), and [MAP NEO 7B](https://huggingface.co/m-a-p/neo_7b).

Reviewer: ```suggestion
We ran a suite of standard benchmarks to compare our model with [Llama 3.1 8B](https://huggingface.co/meta-llama/Meta-Llama-3.1-8B), and the open-source 7-8B models [Olmo 2 7B](https://huggingface.co/allenai/OLMo-2-1124-7B), and [MAP-Neo 7B](https://huggingface.co/m-a-p/neo_7b).
```

For all benchmarks, we used [LM Eval Harness](https://github.com/EleutherAI/lm-evaluation-harness) with the default setup for each task. (These numbers may differ from reported results due to differences in setup. LM Eval Harness is usually somewhat stricter than other harnesses.)

> |                                      | Average  | AGI Eval LSAT-AR | ARC Easy | ARC Challenge | BBH      | BoolQ    | CommonSense QA | COPA     | GPQA     | HellaSwag 0-shot | HellaSwag 10-shot | lambada_openai |  MMLU 5-shot | MMLU 0-shot | MMLU Pro |OpenBookQA | PIQA     | WinoGrande | WSC      |
> |--------------------------------------|----------|------------------|----------|---------------|----------|----------|----------------|----------|----------|------------------|-------------------|----------------|--------------|-------------|----------|-----------|----------|------------|----------|
> | Marin 8B Base <br/>(Deeper Starling) | **68.3** | 20.9             | **86.5** | **63.1**      | **50.6** | **85.9** | 79.1           | **92.0** | 30.3     | **82.3**         | **83.6**          | **74.7**       |  **67.6**    | **65.9**    | **36.5** |44.2       | **84.4** | **74.5**   | 82.1     |
> | Llama 3.1 Base                       | 67.0     | 20.4             | 85.8     | 58.9          | 46.4     | 84.2     | 75.2           | **92.0** | **32.3** | 79.4             | 81.9              | **74.7**       |  66.4        | 65.5        | 33.3     |45.8       | 82.9     | 74.4       | 83.5     |
> | OLMo 2 Base                          | 66.7     | 17.4             | 85.0     | 60.7          | 44.4     | 85.5     | 75.4           | 89.0     | 26.8     | 80.5             | 81.7              | 73.1           |  63.9        | 61.9        | 30.6     |**46.2**   | 82.5     | 74.3       | **86.1** |
> | MAP NEO 7B                           | 62.2     | **23.0**         | 81.1     | 52.0          | 42.4     | 84.7     | **81.7**       | 82.0     | 27.8     | 72.5             | 73.3              | 64.6           |  58.2        | 56.4        | 25.2     |39.4       | 79.0     | 66.1       | 73.3     |

Reviewer 2: ```suggestion
Model | Average | AGI Eval LSAT-AR | ARC Challenge | ARC Easy | BBH | BoolQ | CommonSense QA | COPA | GPQA | GSM8K | HellaSwag_1, 10 shot | HellaSwag, 0 shot | lambada_openai | MMLU Pro | MMLU_5shot | MMLU-0shot | OpenBookQA | PIQA | WinoGrande | WSC
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
Marin 8B Base | 66.6 | 20.9 | 63.1 | 86.5 | 50.6 | 85.9 | 79.1 | 92.0 | 30.3 | 61.3 | 83.6 | 82.3 | 74.7 | 36.5 | 67.6 | 65.9 | 44.2 | 84.4 | 74.5 | 82.1
Llama 3.1 Base | 65.3 | 20.4 | 58.9 | 85.8 | 46.4 | 84.2 | 75.2 | 92.0 | 32.3 | 56.8 | 81.9 | 79.4 | 74.7 | 33.3 | 66.4 | 65.5 | 45.8 | 82.9 | 74.4 | 83.5
OLMo 2 Base | 64.9 | 17.4 | 60.7 | 85.0 | 44.4 | 85.5 | 75.4 | 89.0 | 26.8 | 67.6 | 81.7 | 80.5 | 73.1 | 30.6 | 63.9 | 61.9 | 46.2 | 82.5 | 74.3 | 86.1
MAP NEO 7B | 59.5 | 23.0 | 52.0 | 81.1 | 42.4 | 84.7 | 81.7 | 82.0 | 27.8 | 48.0 | 73.3 | 72.5 | 64.6 | 25.2 | 58.2 | 56.4 | 39.4 | 79.0 | 66.1 | 73.3
Amber 7B | 48.1 | 19.1 | 41.6 | 74.7 | 31.6 | 68.8 | 20.6 | 87.0 | 26.3 | 4.4 | 73.9 | 72.4 | 66.8 | 11.6 | 26.6 | 26.7 | 39.2 | 79.8 | 65.3 | 76.9
```
Model | Average | AGI Eval LSAT-AR | ARC Challenge | ARC Easy | BBH | BoolQ | CommonSense QA | COPA | GPQA | GSM8K | HellaSwag_1, 10 shot | HellaSwag, 0 shot | lambada_openai | MMLU Pro | MMLU_5shot | MMLU-0shot | OpenBookQA | PIQA | WinoGrande | WSC
-- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | --
Marin 8B Base | 66.6 | 20.9 | 63.1 | 86.5 | 50.6 | 85.9 | 79.1 | 92.0 | 30.3 | 61.3 | 83.6 | 82.3 | 74.7 | 36.5 | 67.6 | 65.9 | 44.2 | 84.4 | 74.5 | 82.1
Llama 3.1 Base | 65.3 | 20.4 | 58.9 | 85.8 | 46.4 | 84.2 | 75.2 | 92.0 | 32.3 | 56.8 | 81.9 | 79.4 | 74.7 | 33.3 | 66.4 | 65.5 | 45.8 | 82.9 | 74.4 | 83.5
OLMo 2 Base | 64.9 | 17.4 | 60.7 | 85.0 | 44.4 | 85.5 | 75.4 | 89.0 | 26.8 | 67.6 | 81.7 | 80.5 | 73.1 | 30.6 | 63.9 | 61.9 | 46.2 | 82.5 | 74.3 | 86.1
MAP NEO 7B | 59.5 | 23.0 | 52.0 | 81.1 | 42.4 | 84.7 | 81.7 | 82.0 | 27.8 | 48.0 | 73.3 | 72.5 | 64.6 | 25.2 | 58.2 | 56.4 | 39.4 | 79.0 | 66.1 | 73.3
Amber 7B | 48.1 | 19.1 | 41.6 | 74.7 | 31.6 | 68.8 | 20.6 | 87.0 | 26.3 | 4.4 | 73.9 | 72.4 | 66.8 | 11.6 | 26.6 | 26.7 | 39.2 | 79.8 | 65.3 | 76.9

Reviewer 2: Adds GSM8k

Marin 8B Base fares well on most of these tasks.

## Model Details

Please see [our technical retrospective](https://marin.readthedocs.io/en/latest/reports/marin-8b-retro.html) for more details on the pretraining process.

### Architecture Details

- **Architecture:** Llama 3 8B
- **Hidden size:** 4096
- **Feedforward size:** 14336
- **Number of layers:** 32
- **Number of attention heads:** 32
- **Number of KV heads:** 8

### Tokenizer Details

> Marin 8B uses a variant of the Llama 3 tokenizer: [stanford-crfm/marin-tokenizer](https://huggingface.co/stanford-crfm/marin-tokenizer/). It has the same vocabulary but bundles a chat template into the base tokenizer for convenience.

Reviewer 2: If we use the HuggingFace rename, it'll all redirect. Happy to do this now if you think it's fine David

`All links to this model will automatically redirect to the new location, including git operations. However, to avoid confusion, we recommend updating any existing local clones to point to the new repository URL. To do so, you can use the following command:
`
